src/test/java/StringManipulationTest.java

```diff
@@ -1,149 +1,370 @@
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
+import java.util.Arrays;
 
 import static org.junit.jupiter.api.Assertions.*;
 
 public class StringManipulationTest {
 
-    private StringManipulationInterface manipulatedstring;
+    private StringManipulationInterface manipulatedString;
+
 
     @BeforeEach
     public void setUp() {
-        manipulatedstring = new StringManipulation();
+        manipulatedString = new StringManipulation();
     }
 
     @AfterEach
     public void tearDown() {
-        manipulatedstring = null;
+        manipulatedString = null;
+    }
+
+
+    @Test
+    @DisplayName("Test GetString with a new object")
+    public void testGetString1() {
+        // arrange
+        manipulatedString = new StringManipulation();
+
+        // act
+        String result = manipulatedString.getString();
+
+        // assert
+        assertNull(result);
+    }
+
+    @Test
+    @DisplayName("Test GetString with a basic string")
+    public void testGetString2() {
+        // arrange
+        String expected = "string";
+        manipulatedString.setString(expected);
+
+        // act
+        String actual = manipulatedString.getString();
+
+        // assert
+        assertEquals(expected, actual);
     }
 
+
     @Test
+    @DisplayName("Test Count with a new object")
     public void testCount1() {
-        manipulatedstring.setString("This is my string");
-        int length = manipulatedstring.count();
-        assertEquals(4, length);
+        // arrange
+        manipulatedString = new StringManipulation();
+
+        // act
+        int actual = manipulatedString.count();
+
+        // assert
+        assertEquals(0, actual);
     }
 
     @Test
+    @DisplayName("Test Count with an empty string")
     public void testCount2() {
-        fail("Not yet implemented");
+        // arrange
+        manipulatedString.setString("");
+
+        // act
+        int actual = manipulatedString.count();
+
+        // assert
+        assertEquals(0, actual);
     }
 
     @Test
+    @DisplayName("Test Count with a multi-word string")
     public void testCount3() {
-        fail("Not yet implemented");
+        // arrange
+        manipulatedString.setString("This is my string");
+        int expected = 4;
+
+        // act
+        int actual = manipulatedString.count();
+
+        // assert
+        assertEquals(expected, actual);
     }
 
     @Test
+    @DisplayName("Test Count with trailing spaces on the string")
     public void testCount4() {
-        fail("Not yet implemented");
+        // arrange
+        manipulatedString.setString("Here are spaces    ");
+        int expected = 3;
+
+        // act
+        int actual = manipulatedString.count();
+
+        // assert
+        assertEquals(expected, actual);
```

Comment:
Really small detail I noticed in your written tests is the deceleration of a separate int over just using the number in the parameter, I assume this is an organizational choice due to the comments, weirded me out at first but I can definitely appreciate the tidiness after reading it haha